v-next/hardhat/src/internal/cli/init/init.ts

```diff
@@ -39,6 +40,7 @@ import { getTemplates } from "./template.js";
 
 export interface InitHardhatOptions {
   workspace?: string;
+  esm?: boolean;
```

Comment:
Can we add a name that better describes what this does?

Reply:
I renamed this to `migrateToEsm`.